kubetools/kubernetes/api.py

```diff
@@ -274,11 +274,17 @@ def list_running_jobs(env, namespace):
     return [job for job in jobs if is_running(job)]
 
 
-def delete_job(env, namespace, job):
+valid_propagation_policies = ["Orphan", "Background", "Foreground"]
+
+
+def delete_job(env, namespace, job, propagation_policy=None):
+    if propagation_policy and propagation_policy not in valid_propagation_policies:
```

Comment:
I felt like some sign posting of potential values was useful here but default is a bit tricky as it is config dependent. The underlying api does ignore `None` values though so this does maintain defaults correctly. A _bit_ reliant on client implementation but I thought the added 'documentation' was worth the associated risk